components/thanks_io/sources/common/common.mjs

```diff
@@ -0,0 +1,54 @@
+import thanksIo from "../../thanks_io.app.mjs";
+
+export default {
+  props: {
+    thanksIo,
+    db: "$.service.db",
+    timer: {
+      label: "Polling interval",
+      description: "Pipedream will poll the API on this schedule",
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: 60 * 15,
+      },
+    },
+    subAccount: {
+      propDefinition: [
+        thanksIo,
+        "subAccount",
+      ],
+    },
+  },
+  methods: {
+    _getLastTimestamp() {
+      return this.db.get("lastTimestamp") || null;
+    },
+    _setLastTimestamp(lastTimestamp) {
+      this.db.set("lastTimestamp", lastTimestamp);
+    },
+    isLater(newDate, compareDate) {
+      return compareDate
+        ? (newDate > compareDate)
+        : true;
+    },
+    async paginate(resourceFn, params) {
+      let next;
+      const items = [];
+      do {
+        const {
+          data, links,
+        } = next
+          ? await this._makeRequest({
+            url: next,
+            params,
+          })
+          : await resourceFn({
+            params,
+          });
```

Comment:
With the modification I suggested previously to `_makeRequest()` function you can just do:
```suggestion
        const {
          data, links,
        } = await resourceFn({
            url: next,
            params,
        });
```